File: rael/0401/PGS_모음사전.java
class Solution {
	
    static int idx = 0;
    static int answer = -1;
	
    public int solution(String word) {   	
        dfs(word, "");        
        return answer;
    }
    
    public void dfs(String word, String text) {
    	if(answer > 0) return;
    	if(word.equals(text)) {
    		answer=idx;
    	}
    	idx++;
    	if(text.length()==5) {    		
    		return;
    	}
    	
    	dfs(word, text+"A");
    	dfs(word, text+"E");
    	dfs(word, text+"I");
    	dfs(word, text+"O");
    	dfs(word, text+"U");
    }

}

File: woojae/2024_5/0501/PGS_모음사전.py
alphabets = ["A", "E", "I", "O", "U"]
answer = []

def search(target, word):
    answer.append(["", 0] if not answer else [target, answer[-1][1]+1])
    if target == word or len(target) == 5:
        return
    
    for alphabet in alphabets:
        if answer[-1][0] != word:
            search(target + alphabet, word)
        else:
            return


def solution(word):
    search("", word)
    return answer[-1][1]
